diff --git a/core/libcutils/canned_fs_config.cpp b/core/libcutils/canned_fs_config.cpp
index b677949a..0e0f8714 100644
--- a/core/libcutils/canned_fs_config.cpp
+++ b/core/libcutils/canned_fs_config.cpp
@@ -86,7 +86,7 @@ int load_canned_fs_config(const char* fn) {
             // Historical: there can be tokens like "selabel=..." here. They have been ignored.
             // It's not an error because selabels are applied separately in e2fsdroid using the
             // file_contexts files set via -S option.
-            std::cerr << "info: ignored token \"" << sv << "\" in " << fn << std::endl;
+            // std::cerr << "info: ignored token \"" << sv << "\" in " << fn << std::endl;
         }
 
         canned_data.emplace_back(std::move(e));
