diff --git a/mkbootfs/mkbootfs.cpp b/mkbootfs/mkbootfs.cpp
index a45c6a20..1c8a3726 100644
--- a/mkbootfs/mkbootfs.cpp
+++ b/mkbootfs/mkbootfs.cpp
@@ -164,10 +164,10 @@ static void _archive_dir(char *in, char *out, int ilen, int olen)
 
     while((de = readdir(d)) != 0){
             /* xxx: feature? maybe some dotfiles are okay */
-        if(de->d_name[0] == '.') continue;
+        if(!strcmp(de->d_name, ".") || !strcmp(de->d_name, "..")) continue;
 
             /* xxx: hack. use a real exclude list */
-        if(!strcmp(de->d_name, "root")) continue;
+        // if(!strcmp(de->d_name, "root")) continue;
 
         if (entries >= size) {
           size *= 2;
